src/libsyntax_ext/lib.rs

```diff
@@ -42,17 +42,17 @@ pub mod proc_macro_impl;
 use rustc_data_structures::sync::Lrc;
 use syntax::ast;
 use syntax::ext::base::{MacroExpanderFn, NormalTT, NamedSyntaxExtension, MultiModifier};
-use syntax::ext::hygiene;
 use syntax::symbol::Symbol;
 
 pub fn register_builtins(resolver: &mut dyn syntax::ext::base::Resolver,
                          user_exts: Vec<NamedSyntaxExtension>) {
     deriving::register_builtin_derives(resolver);
 
+    let edition = resolver.parse_sess().edition;
```

Comment:
Why not adding the edition parameter to `fn register_builtins`?
Parse session is available from where `register_builtins` is called.
`Resolver::parse_sess` would be unnecessary then.